Yank/restraints.py

```diff
@@ -1914,26 +1844,62 @@ def compute_atom_str(input_string, topography_key):
             raise ValueError(error_msg.format(min_distance, max_distance))
         r3_l1_pairs = r3_l1_pairs[indices_of_in_range_pairs].tolist()
 
+        def find_bonded_to(input_atom_index, comparison_set):
```

Comment:
It isn't easier to just iterate through forcefield `HarmonicAngleForce` terms? The advantage of doing that would also be that we would be guaranteed none of those would have `theta0 = 0` or `180 degrees`, while just looking at three bonded atoms could mean they are still collinear.

Reply:
I thought about that, and that would require a a different enumeration through the thermodynamic_state -> system -> forces -> harmonic angle force. Although this internal function does not accept a thermodynamic state, the only time its called is from the outer function which does, so I can make that change pretty easily, this was just working with what I already had to try and minimize the structural changes.

If you think that is a better option, I can rework it

Reply:
What do you think @andrrizzi?

Reply:
If it further cuts the number of lines of code, I'm +1.